zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/deployment/DeploymentRecord.java

```diff
@@ -193,10 +215,22 @@ public boolean hasForms() {
         .anyMatch(x -> x.endsWith(".form"));
   }
 
+  public boolean hasResources() {
+    return getResources().stream()
+        .map(io.camunda.zeebe.protocol.record.value.deployment.DeploymentResource::getResourceName)
+        .anyMatch(
+            x ->
+                !x.endsWith(".form")
+                    && !x.endsWith(".xml")
+                    && !x.endsWith(".bpmn")
+                    && !x.endsWith(".dmn"));
+  }
```

Comment:
❌  This method will allow any type of resource other than a Form, BPMN, DMN, or XML file. We also need something to check for robot script files before transforming them.